### PR TITLE
Return Error When the Method Pointer Not Found

### LINKED ISSUE #1082: Language Server Does Not Send `executionContext/expressionValuesComputed`
### General Summary
The language server is not sending any  `executionContext/expressionValuesComputed` notifications.

### Steps to Reproduce
Run IDE with the following `Main.enso`
```

Text.overloaded arg = 10
Number.overloaded arg = 20

main = 
    local_fun a = 30
    val1 = 15.overloaded 1 
    "foo".overloaded 2
    overloaded 10 30
    Unit


#### METADATA ####
[[{"index":{"value":1},"size":{"value":4}},"9b398aba-9c84-4fdf-bb11-df1605781370"],[{"index":{"value":5},"size":{"value":1}},"1294ca97-fde1-4c1d-a318-2e5574332d1c"],[{"index":{"value":6},"size":{"value":10}},"de64ff91-5028-4e54-932e-6801b798d746"],[{"index":{"value":1},"size":{"value":15}},"be784ae5-fcfe-41c8-bb9f-f4e0a6560acd"],[{"index":{"value":17},"size":{"value":3}},"4542303b-94d5-469f-b539-3d0c635bec04"],[{"index":{"value":1},"size":{"value":19}},"6c9c6a52-c9a3-48ef-b26f-b01baebac399"],[{"index":{"value":21},"size":{"value":1}},"38105ed5-bfd6-46d1-9db5-e38d771611a9"],[{"index":{"value":23},"size":{"value":2}},"2f4b9ec8-ae60-4b09-ad00-dad434c32fbf"],[{"index":{"value":1},"size":{"value":24}},"89306fc6-4d37-4bf2-8fd6-802cc31a4c13"],[{"index":{"value":26},"size":{"value":6}},"0e6eaadc-ceaa-406b-ba0a-364d233c5991"],[{"index":{"value":32},"size":{"value":1}},"a55dafec-accd-4d59-8bc7-4c27259a5103"],[{"index":{"value":33},"size":{"value":10}},"e99b351f-08ce-435b-bc51-5f7fca49e58d"],[{"index":{"value":26},"size":{"value":17}},"511979ea-a9cd-4b4e-bd2f-7dfb5efbb2b7"],[{"index":{"value":44},"size":{"value":3}},"dfa2ac97-6eae-4db3-90a5-87121e7d2eb4"],[{"index":{"value":26},"size":{"value":21}},"b7e72872-ea0a-4946-a65c-ae002140200e"],[{"index":{"value":48},"size":{"value":1}},"173c8896-df0c-41e1-970e-a8c189f3da51"],[{"index":{"value":50},"size":{"value":2}},"c67cc568-ca5a-4fca-8840-e879a5fbc57f"],[{"index":{"value":26},"size":{"value":26}},"83cd373f-925c-4e2d-8793-7e6df70e5146"],[{"index":{"value":54},"size":{"value":4}},"b3237818-d1ae-4ccc-870f-e0f78d46f7f4"],[{"index":{"value":59},"size":{"value":1}},"8d2164cb-cb87-4cfb-be62-1f40d3c05045"],[{"index":{"value":66},"size":{"value":9}},"4ac46f04-370d-4ada-bfaa-b016e0066446"],[{"index":{"value":76},"size":{"value":1}},"92d89152-f80e-433c-8c33-8ef9ff8cbae5"],[{"index":{"value":66},"size":{"value":11}},"1022a197-9e9b-4c22-a4da-2b313bcd4567"],[{"index":{"value":78},"size":{"value":1}},"112c49b1-76f9-4466-a631-81b03a85a830"],[{"index":{"value":80},"size":{"value":2}},"0d629577-b1ac-40aa-bc8c-4f623f0ee9fd"],[{"index":{"value":66},"size":{"value":16}},"aafeabbd-3b3e-4cff-9443-8fb4c00a75a9"],[{"index":{"value":87},"size":{"value":4}},"1be68729-ac0e-4f42-bd9f-64496bac0549"],[{"index":{"value":92},"size":{"value":1}},"dfeac92f-548d-4850-9997-663d17fb0b2c"],[{"index":{"value":94},"size":{"value":2}},"ce633f0d-3718-4b1c-b2c1-c0b4c4040cbd"],[{"index":{"value":96},"size":{"value":1}},"36e2a34c-b46e-4b09-9ae2-4952637f76b5"],[{"index":{"value":97},"size":{"value":10}},"c1e6df06-9c18-4397-acd1-0b7f48c29eed"],[{"index":{"value":94},"size":{"value":13}},"311834f9-5178-42b0-a1cc-c374f3040f3b"],[{"index":{"value":108},"size":{"value":1}},"5d4b06e7-de0a-4122-adb1-2aeb8cff4fc4"],[{"index":{"value":94},"size":{"value":15}},"3f7f93f9-36c2-4841-bed4-3230ff69e824"],[{"index":{"value":87},"size":{"value":22}},"2c6b08d6-346d-4ef5-ad0f-9b4b8fa674b2"],[{"index":{"value":115},"size":{"value":5}},"241dcd02-57a5-4d59-b10d-c744bf9b308b"],[{"index":{"value":120},"size":{"value":1}},"7763d344-706e-4314-ad41-b2e523e4b306"],[{"index":{"value":121},"size":{"value":10}},"5740483d-a178-466e-aec8-77d4f2ec36ac"],[{"index":{"value":115},"size":{"value":16}},"18484752-ff83-4762-ae76-00d558487d95"],[{"index":{"value":132},"size":{"value":1}},"425ca866-6d9f-4e67-8031-a45edcbe240a"],[{"index":{"value":115},"size":{"value":18}},"1e0e5497-883c-4280-a881-ac7c11b6ae93"],[{"index":{"value":138},"size":{"value":10}},"93959f0d-5d92-4ed1-a59e-a8b085fd9889"],[{"index":{"value":149},"size":{"value":2}},"50022a96-15fe-47a5-ab67-afc839d14fba"],[{"index":{"value":138},"size":{"value":13}},"bbe232f5-f573-4ffb-b478-efcad6787618"],[{"index":{"value":152},"size":{"value":2}},"ec1a9979-7c84-473e-b1ed-0874332af6e5"],[{"index":{"value":138},"size":{"value":16}},"dc655e43-ad57-4d28-814c-9155b115b560"],[{"index":{"value":159},"size":{"value":4}},"e2ceb34b-beb8-4b3b-8a6a-0cd1493f6241"],[{"index":{"value":61},"size":{"value":102}},"00b72233-0231-4871-8f14-5adbedb86d3a"],[{"index":{"value":54},"size":{"value":109}},"14c0d582-4f26-4fb5-934f-5d1e8afac796"]]
{"ide":{"node":{"2c6b08d6-346d-4ef5-ad0f-9b4b8fa674b2":{"position":{"vector":[-552.0,22.0]},"intended_method":null},"3f7f93f9-36c2-4841-bed4-3230ff69e824":{"position":{"vector":[-471.0,20.999735]},"intended_method":null},"1e0e5497-883c-4280-a881-ac7c11b6ae93":{"position":{"vector":[-448.0,98.05282]},"intended_method":null},"dc655e43-ad57-4d28-814c-9155b115b560":{"position":{"vector":[-291.0,48.026684]},"intended_method":null},"e2ceb34b-beb8-4b3b-8a6a-0cd1493f6241":{"position":{"vector":[-604.0,99.0]},"intended_method":null}}}}
```

### Expected Result
Engine eventually sends `executionContext/expressionValuesComputed` with updates for all the expressions in the main method.

### Actual Result
Engine does not send any `executionContext/expressionValuesComputed`  notification.

See the socket log:
```
{"jsonrpc":"2.0","id":0,"method":"session/initProtocolConnection","params":{"clientId":"9fa35949-ec1b-454d-bb39-97af038b7af8"}} | 127 | 15:38:52.169
{"jsonrpc":"2.0","id":0,"result":{"contentRoots":["21e8425f-2b9f-49c1-9ca8-57409f44d7b3"]}} | 91 | 15:38:52.201
{"jsonrpc":"2.0","id":1,"method":"search/getSuggestionsDatabase","params":{}} | 77 | 15:38:52.235
{"jsonrpc":"2.0","id":1,"result":{"entries":[{"id":166,"suggestion":{"type":"local","externalId":"1211e45d-756b-463d-98b5-fa5e964748d2","module":"Unnamed.Main","name":"result","returnType":"Any","scope":{"start":{"line":2,"character":6},"end":{"line":4,"character":5}}}},{"id":168,"suggestion":{"type":"method","externalId":"c67cc568-ca5a-4fca-8840-e879a5fbc57f","module":"Test.Main","name":"overloaded","arguments":[{"name":"this","reprType":"Any","isSuspended":false,"hasDefault":false,"defaultValue":null},{"name":"arg","reprType":"Any","isSuspended":false,"hasDefault":false,"defaultValue":null}],"selfType":"Number","returnType":"Any"}},{"id":171,"suggestion":{"type":"local","externalId":"3f7f93f9-36c2-4841-bed4-3230ff69e824","module":"Test.Main","name":"val1","returnType":"Any","scope":{"start":{"line":4,"character":7},"end":{"line":9,"character":8}}}},{"id":167,"suggestion":{"type":"method","externalId":"2f4b9ec8-ae60-4b09-ad00-dad434c32fbf","module":"Test.Main","name":"overloaded","arguments":[{"name":"this","reprType":"Any","isSuspended":false,"hasDefault":false,"defaultValue":null},{"name":"arg","reprType":"Any","isSuspended":false,"hasDefault":false,"defaultValue":null}],"selfType":"Text","returnType":"Any"}},{"id":169,"suggestion":{"type":"method","externalId":"00b72233-0231-4871-8f14-5adbedb86d3a","module":"Test.Main","name":"main","arguments":[{"name":"this","reprType":"Any","isSuspended":false,"hasDefault":false,"defaultValue":null}],"selfType":"here","returnType":"Any"}},{"id":164,"suggestion":{"type":"method","externalId":"4ecc8aff-db16-44c4-afa3-717583cbe0a5","module":"Unnamed.Main","name":"mulAdd","arguments":[{"name":"this","reprType":"Any","isSuspended":false,"hasDefault":false,"defaultValue":null}],"selfType":"Main","returnType":"Any"}},{"id":165,"suggestion":{"type":"method","externalId":"c6f005bf-8afe-4af2-bc34-266e5999e3f3","module":"Unnamed.Main","name":"main","arguments":[{"name":"this","reprType":"Any","isSuspended":false,"hasDefault":false,"defaultValue":null}],"selfType":"here","returnType":"Number"}},{"id":170,"suggestion":{"type":"function","externalId":"0d629577-b1ac-40aa-bc8c-4f623f0ee9fd","module":"Test.Main","name":"local_fun","arguments":[{"name":"a","reprType":"Any","isSuspended":false,"hasDefault":false,"defaultValue":null}],"returnType":"Any","scope":{"start":{"line":4,"character":7},"end":{"line":9,"character":8}}}}],"currentVersion":527}} | 2414 | 15:38:52.485
{"jsonrpc":"2.0","id":2,"method":"capability/acquire","params":{"method":"search/receivesSuggestionsDatabaseUpdates","registerOptions":{}}} | 139 | 15:38:52.506
{"jsonrpc":"2.0","id":2,"result":null} | 38 | 15:38:52.512
{"jsonrpc":"2.0","id":3,"method":"file/exists","params":{"path":{"rootId":"21e8425f-2b9f-49c1-9ca8-57409f44d7b3","segments":["src","Main.enso"]}}} | 146 | 15:38:52.517
{"jsonrpc":"2.0","id":3,"result":{"exists":true}} | 49 | 15:38:52.604
{"jsonrpc":"2.0","id":4,"method":"text/openFile","params":{"path":{"rootId":"21e8425f-2b9f-49c1-9ca8-57409f44d7b3","segments":["src","Main.enso"]}}} | 148 | 15:38:52.619
{"jsonrpc":"2.0","id":4,"result":{"writeCapability":{"method":"text/canEdit","registerOptions":{"path":{"rootId":"21e8425f-2b9f-49c1-9ca8-57409f44d7b3","segments":["src","Main.enso"]}}},"content":"\nText.overloaded arg = 10\nNumber.overloaded arg = 20\n\nmain = \n local_fun a = 30\n val1 = 15.overloaded 1 \n \"foo\".overloaded 2\n overloaded 10 30\n Unit\n\n\n#### METADATA ####\n[[{\"index\":{\"value\":1},\"size\":{\"value\":4}},\"9b398aba-9c84-4fdf-bb11-df1605781370\"],[{\"index\":{\"value\":5},\"size\":{\"value\":1}},\"1294ca97-fde1-4c1d-a318-2e5574332d1c\"],[{\"index\":{\"value\":6},\"size\":{\"value\":10}},\"de64ff91-5028-4e54-932e-6801b798d746\"],[{\"index\":{\"value\":1},\"size\":{\"value\":15}},\"be784ae5-fcfe-41c8-bb9f-f4e0a6560acd\"],[{\"index\":{\"value\":17},\"size\":{\"value\":3}},\"4542303b-94d5-469f-b539-3d0c635bec04\"],[{\"index\":{\"value\":1},\"size\":{\"value\":19}},\"6c9c6a52-c9a3-48ef-b26f-b01baebac399\"],[{\"index\":{\"value\":21},\"size\":{\"value\":1}},\"38105ed5-bfd6-46d1-9db5-e38d771611a9\"],[{\"index\":{\"value\":23},\"size\":{\"value\":2}},\"2f4b9ec8-ae60-4b09-ad00-dad434c32fbf\"],[{\"index\":{\"value\":1},\"size\":{\"value\":24}},\"89306fc6-4d37-4bf2-8fd6-802cc31a4c13\"],[{\"index\":{\"value\":26},\"size\":{\"value\":6}},\"0e6eaadc-ceaa-406b-ba0a-364d233c5991\"],[{\"index\":{\"value\":32},\"size\":{\"value\":1}},\"a55dafec-accd-4d59-8bc7-4c27259a5103\"],[{\"index\":{\"value\":33},\"size\":{\"value\":10}},\"e99b351f-08ce-435b-bc51-5f7fca49e58d\"],[{\"index\":{\"value\":26},\"size\":{\"value\":17}},\"511979ea-a9cd-4b4e-bd2f-7dfb5efbb2b7\"],[{\"index\":{\"value\":44},\"size\":{\"value\":3}},\"dfa2ac97-6eae-4db3-90a5-87121e7d2eb4\"],[{\"index\":{\"value\":26},\"size\":{\"value\":21}},\"b7e72872-ea0a-4946-a65c-ae002140200e\"],[{\"index\":{\"value\":48},\"size\":{\"value\":1}},\"173c8896-df0c-41e1-970e-a8c189f3da51\"],[{\"index\":{\"value\":50},\"size\":{\"value\":2}},\"c67cc568-ca5a-4fca-8840-e879a5fbc57f\"],[{\"index\":{\"value\":26},\"size\":{\"value\":26}},\"83cd373f-925c-4e2d-8793-7e6df70e5146\"],[{\"index\":{\"value\":54},\"size\":{\"value\":4}},\"b3237818-d1ae-4ccc-870f-e0f78d46f7f4\"],[{\"index\":{\"value\":59},\"size\":{\"value\":1}},\"8d2164cb-cb87-4cfb-be62-1f40d3c05045\"],[{\"index\":{\"value\":66},\"size\":{\"value\":9}},\"4ac46f04-370d-4ada-bfaa-b016e0066446\"],[{\"index\":{\"value\":76},\"size\":{\"value\":1}},\"92d89152-f80e-433c-8c33-8ef9ff8cbae5\"],[{\"index\":{\"value\":66},\"size\":{\"value\":11}},\"1022a197-9e9b-4c22-a4da-2b313bcd4567\"],[{\"index\":{\"value\":78},\"size\":{\"value\":1}},\"112c49b1-76f9-4466-a631-81b03a85a830\"],[{\"index\":{\"value\":80},\"size\":{\"value\":2}},\"0d629577-b1ac-40aa-bc8c-4f623f0ee9fd\"],[{\"index\":{\"value\":66},\"size\":{\"value\":16}},\"aafeabbd-3b3e-4cff-9443-8fb4c00a75a9\"],[{\"index\":{\"value\":87},\"size\":{\"value\":4}},\"1be68729-ac0e-4f42-bd9f-64496bac0549\"],[{\"index\":{\"value\":92},\"size\":{\"value\":1}},\"dfeac92f-548d-4850-9997-663d17fb0b2c\"],[{\"index\":{\"value\":94},\"size\":{\"value\":2}},\"ce633f0d-3718-4b1c-b2c1-c0b4c4040cbd\"],[{\"index\":{\"value\":96},\"size\":{\"value\":1}},\"36e2a34c-b46e-4b09-9ae2-4952637f76b5\"],[{\"index\":{\"value\":97},\"size\":{\"value\":10}},\"c1e6df06-9c18-4397-acd1-0b7f48c29eed\"],[{\"index\":{\"value\":94},\"size\":{\"value\":13}},\"311834f9-5178-42b0-a1cc-c374f3040f3b\"],[{\"index\":{\"value\":108},\"size\":{\"value\":1}},\"5d4b06e7-de0a-4122-adb1-2aeb8cff4fc4\"],[{\"index\":{\"value\":94},\"size\":{\"value\":15}},\"3f7f93f9-36c2-4841-bed4-3230ff69e824\"],[{\"index\":{\"value\":87},\"size\":{\"value\":22}},\"2c6b08d6-346d-4ef5-ad0f-9b4b8fa674b2\"],[{\"index\":{\"value\":115},\"size\":{\"value\":5}},\"241dcd02-57a5-4d59-b10d-c744bf9b308b\"],[{\"index\":{\"value\":120},\"size\":{\"value\":1}},\"7763d344-706e-4314-ad41-b2e523e4b306\"],[{\"index\":{\"value\":121},\"size\":{\"value\":10}},\"5740483d-a178-466e-aec8-77d4f2ec36ac\"],[{\"index\":{\"value\":115},\"size\":{\"value\":16}},\"18484752-ff83-4762-ae76-00d558487d95\"],[{\"index\":{\"value\":132},\"size\":{\"value\":1}},\"425ca866-6d9f-4e67-8031-a45edcbe240a\"],[{\"index\":{\"value\":115},\"size\":{\"value\":18}},\"1e0e5497-883c-4280-a881-ac7c11b6ae93\"],[{\"index\":{\"value\":138},\"size\":{\"value\":10}},\"93959f0d-5d92-4ed1-a59e-a8b085fd9889\"],[{\"index\":{\"value\":149},\"size\":{\"value\":2}},\"50022a96-15fe-47a5-ab67-afc839d14fba\"],[{\"index\":{\"value\":138},\"size\":{\"value\":13}},\"bbe232f5-f573-4ffb-b478-efcad6787618\"],[{\"index\":{\"value\":152},\"size\":{\"value\":2}},\"ec1a9979-7c84-473e-b1ed-0874332af6e5\"],[{\"index\":{\"value\":138},\"size\":{\"value\":16}},\"dc655e43-ad57-4d28-814c-9155b115b560\"],[{\"index\":{\"value\":159},\"size\":{\"value\":4}},\"e2ceb34b-beb8-4b3b-8a6a-0cd1493f6241\"],[{\"index\":{\"value\":61},\"size\":{\"value\":102}},\"00b72233-0231-4871-8f14-5adbedb86d3a\"],[{\"index\":{\"value\":54},\"size\":{\"value\":109}},\"14c0d582-4f26-4fb5-934f-5d1e8afac796\"]]\n{\"ide\":{\"node\":{\"2c6b08d6-346d-4ef5-ad0f-9b4b8fa674b2\":{\"position\":{\"vector\":[-552.0,22.0]},\"intended_method\":null},\"3f7f93f9-36c2-4841-bed4-3230ff69e824\":{\"position\":{\"vector\":[-471.0,20.999735]},\"intended_method\":null},\"1e0e5497-883c-4280-a881-ac7c11b6ae93\":{\"position\":{\"vector\":[-448.0,98.05282]},\"intended_method\":null},\"dc655e43-ad57-4d28-814c-9155b115b560\":{\"position\":{\"vector\":[-291.0,48.026684]},\"intended_method\":null},\"e2ceb34b-beb8-4b3b-8a6a-0cd1493f6241\":{\"position\":{\"vector\":[-604.0,99.0]},\"intended_method\":null}}}}\n\n","currentVersion":"1cfc5004e594bb32952fdd27352cb1fac918071506ff8748c02e8eb3"}} | 5647 | 15:38:52.658
{"jsonrpc":"2.0","id":5,"method":"executionContext/create","params":{}} | 71 | 15:38:52.952
{"jsonrpc":"2.0","id":6,"method":"text/applyEdit","params":{"edit":{"edits":[{"range":{"end":{"character":0,"line":16},"start":{"character":0,"line":0}},"text":"\nText.overloaded arg = 10\nNumber.overloaded arg = 20\n\nmain = \n local_fun a = 30\n val1 = 15.overloaded 1 \n \"foo\".overloaded 2\n overloaded 10 30\n Unit\n\n\n#### METADATA ####\n[[{\"index\":{\"value\":1},\"size\":{\"value\":4}},\"9b398aba-9c84-4fdf-bb11-df1605781370\"],[{\"index\":{\"value\":5},\"size\":{\"value\":1}},\"1294ca97-fde1-4c1d-a318-2e5574332d1c\"],[{\"index\":{\"value\":6},\"size\":{\"value\":10}},\"de64ff91-5028-4e54-932e-6801b798d746\"],[{\"index\":{\"value\":1},\"size\":{\"value\":15}},\"be784ae5-fcfe-41c8-bb9f-f4e0a6560acd\"],[{\"index\":{\"value\":17},\"size\":{\"value\":3}},\"4542303b-94d5-469f-b539-3d0c635bec04\"],[{\"index\":{\"value\":1},\"size\":{\"value\":19}},\"6c9c6a52-c9a3-48ef-b26f-b01baebac399\"],[{\"index\":{\"value\":21},\"size\":{\"value\":1}},\"38105ed5-bfd6-46d1-9db5-e38d771611a9\"],[{\"index\":{\"value\":23},\"size\":{\"value\":2}},\"2f4b9ec8-ae60-4b09-ad00-dad434c32fbf\"],[{\"index\":{\"value\":1},\"size\":{\"value\":24}},\"89306fc6-4d37-4bf2-8fd6-802cc31a4c13\"],[{\"index\":{\"value\":26},\"size\":{\"value\":6}},\"0e6eaadc-ceaa-406b-ba0a-364d233c5991\"],[{\"index\":{\"value\":32},\"size\":{\"value\":1}},\"a55dafec-accd-4d59-8bc7-4c27259a5103\"],[{\"index\":{\"value\":33},\"size\":{\"value\":10}},\"e99b351f-08ce-435b-bc51-5f7fca49e58d\"],[{\"index\":{\"value\":26},\"size\":{\"value\":17}},\"511979ea-a9cd-4b4e-bd2f-7dfb5efbb2b7\"],[{\"index\":{\"value\":44},\"size\":{\"value\":3}},\"dfa2ac97-6eae-4db3-90a5-87121e7d2eb4\"],[{\"index\":{\"value\":26},\"size\":{\"value\":21}},\"b7e72872-ea0a-4946-a65c-ae002140200e\"],[{\"index\":{\"value\":48},\"size\":{\"value\":1}},\"173c8896-df0c-41e1-970e-a8c189f3da51\"],[{\"index\":{\"value\":50},\"size\":{\"value\":2}},\"c67cc568-ca5a-4fca-8840-e879a5fbc57f\"],[{\"index\":{\"value\":26},\"size\":{\"value\":26}},\"83cd373f-925c-4e2d-8793-7e6df70e5146\"],[{\"index\":{\"value\":54},\"size\":{\"value\":4}},\"b3237818-d1ae-4ccc-870f-e0f78d46f7f4\"],[{\"index\":{\"value\":59},\"size\":{\"value\":1}},\"8d2164cb-cb87-4cfb-be62-1f40d3c05045\"],[{\"index\":{\"value\":66},\"size\":{\"value\":9}},\"4ac46f04-370d-4ada-bfaa-b016e0066446\"],[{\"index\":{\"value\":76},\"size\":{\"value\":1}},\"92d89152-f80e-433c-8c33-8ef9ff8cbae5\"],[{\"index\":{\"value\":66},\"size\":{\"value\":11}},\"1022a197-9e9b-4c22-a4da-2b313bcd4567\"],[{\"index\":{\"value\":78},\"size\":{\"value\":1}},\"112c49b1-76f9-4466-a631-81b03a85a830\"],[{\"index\":{\"value\":80},\"size\":{\"value\":2}},\"0d629577-b1ac-40aa-bc8c-4f623f0ee9fd\"],[{\"index\":{\"value\":66},\"size\":{\"value\":16}},\"aafeabbd-3b3e-4cff-9443-8fb4c00a75a9\"],[{\"index\":{\"value\":87},\"size\":{\"value\":4}},\"1be68729-ac0e-4f42-bd9f-64496bac0549\"],[{\"index\":{\"value\":92},\"size\":{\"value\":1}},\"dfeac92f-548d-4850-9997-663d17fb0b2c\"],[{\"index\":{\"value\":94},\"size\":{\"value\":2}},\"ce633f0d-3718-4b1c-b2c1-c0b4c4040cbd\"],[{\"index\":{\"value\":96},\"size\":{\"value\":1}},\"36e2a34c-b46e-4b09-9ae2-4952637f76b5\"],[{\"index\":{\"value\":97},\"size\":{\"value\":10}},\"c1e6df06-9c18-4397-acd1-0b7f48c29eed\"],[{\"index\":{\"value\":94},\"size\":{\"value\":13}},\"311834f9-5178-42b0-a1cc-c374f3040f3b\"],[{\"index\":{\"value\":108},\"size\":{\"value\":1}},\"5d4b06e7-de0a-4122-adb1-2aeb8cff4fc4\"],[{\"index\":{\"value\":94},\"size\":{\"value\":15}},\"3f7f93f9-36c2-4841-bed4-3230ff69e824\"],[{\"index\":{\"value\":87},\"size\":{\"value\":22}},\"2c6b08d6-346d-4ef5-ad0f-9b4b8fa674b2\"],[{\"index\":{\"value\":115},\"size\":{\"value\":5}},\"241dcd02-57a5-4d59-b10d-c744bf9b308b\"],[{\"index\":{\"value\":120},\"size\":{\"value\":1}},\"7763d344-706e-4314-ad41-b2e523e4b306\"],[{\"index\":{\"value\":121},\"size\":{\"value\":10}},\"5740483d-a178-466e-aec8-77d4f2ec36ac\"],[{\"index\":{\"value\":115},\"size\":{\"value\":16}},\"18484752-ff83-4762-ae76-00d558487d95\"],[{\"index\":{\"value\":132},\"size\":{\"value\":1}},\"425ca866-6d9f-4e67-8031-a45edcbe240a\"],[{\"index\":{\"value\":115},\"size\":{\"value\":18}},\"1e0e5497-883c-4280-a881-ac7c11b6ae93\"],[{\"index\":{\"value\":138},\"size\":{\"value\":10}},\"93959f0d-5d92-4ed1-a59e-a8b085fd9889\"],[{\"index\":{\"value\":149},\"size\":{\"value\":2}},\"50022a96-15fe-47a5-ab67-afc839d14fba\"],[{\"index\":{\"value\":138},\"size\":{\"value\":13}},\"bbe232f5-f573-4ffb-b478-efcad6787618\"],[{\"index\":{\"value\":152},\"size\":{\"value\":2}},\"ec1a9979-7c84-473e-b1ed-0874332af6e5\"],[{\"index\":{\"value\":138},\"size\":{\"value\":16}},\"dc655e43-ad57-4d28-814c-9155b115b560\"],[{\"index\":{\"value\":159},\"size\":{\"value\":4}},\"e2ceb34b-beb8-4b3b-8a6a-0cd1493f6241\"],[{\"index\":{\"value\":61},\"size\":{\"value\":102}},\"00b72233-0231-4871-8f14-5adbedb86d3a\"],[{\"index\":{\"value\":54},\"size\":{\"value\":109}},\"14c0d582-4f26-4fb5-934f-5d1e8afac796\"]]\n{\"ide\":{\"node\":{\"2c6b08d6-346d-4ef5-ad0f-9b4b8fa674b2\":{\"position\":{\"vector\":[-552.0,22.0]},\"intended_method\":null},\"3f7f93f9-36c2-4841-bed4-3230ff69e824\":{\"position\":{\"vector\":[-471.0,20.999735]},\"intended_method\":null},\"1e0e5497-883c-4280-a881-ac7c11b6ae93\":{\"position\":{\"vector\":[-448.0,98.05282]},\"intended_method\":null},\"dc655e43-ad57-4d28-814c-9155b115b560\":{\"position\":{\"vector\":[-291.0,48.026684]},\"intended_method\":null},\"e2ceb34b-beb8-4b3b-8a6a-0cd1493f6241\":{\"position\":{\"vector\":[-604.0,99.0]},\"intended_method\":null}}}}"}],"newVersion":"439e3bbf19e1934d7c72318457949c05bb99929035ce74bb14d8eeaa","oldVersion":"1cfc5004e594bb32952fdd27352cb1fac918071506ff8748c02e8eb3","path":{"rootId":"21e8425f-2b9f-49c1-9ca8-57409f44d7b3","segments":["src","Main.enso"]}}}} | 5766 | 15:38:53.009
{"jsonrpc":"2.0","id":5,"result":{"contextId":"106d0c7b-bf5c-49ee-b018-4c849dc0c8c1","canModify":{"method":"executionContext/canModify","registerOptions":{"contextId":"106d0c7b-bf5c-49ee-b018-4c849dc0c8c1"}},"receivesUpdates":{"method":"executionContext/receivesUpdates","registerOptions":{"contextId":"106d0c7b-bf5c-49ee-b018-4c849dc0c8c1"}}}} | 344 | 15:38:53.010
{"jsonrpc":"2.0","id":7,"method":"executionContext/push","params":{"contextId":"106d0c7b-bf5c-49ee-b018-4c849dc0c8c1","stackItem":{"methodPointer":{"definedOnType":"Main","module":"Unnamed.Main","name":"main"},"positionalArgumentsExpressions":[],"thisArgumentExpression":null,"type":"ExplicitCall"}}} | 300 | 15:38:53.021
{"jsonrpc":"2.0","id":7,"result":null} | 38 | 15:38:53.108
{"jsonrpc":"2.0","id":8,"method":"file/exists","params":{"path":{"rootId":"21e8425f-2b9f-49c1-9ca8-57409f44d7b3","segments":["visualization"]}}} | 144 | 15:38:56.899
{"jsonrpc":"2.0","id":6,"result":null} | 38 | 15:38:57.391
{"jsonrpc":"2.0","id":8,"result":{"exists":false}} | 50 | 15:38:59.266
```

### Enso Version
205c4e0002fad3ea8cb98f7cad00db29b8d8287c
```
Enso Project Manager
Version:    0.1.0
Built with: scala-2.13.3 for GraalVM 20.1.0
Built from: main* @ 205c4e0002fad3ea8cb98f7cad00db29b8d8287c
Running on: OpenJDK 64-Bit Server VM, GraalVM Community, JDK 11.0.7+10-jvmci-20.1-b02
            Windows 10 10.0 (amd64)
```

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

close #1082 

PR enables explicit error notifications when the execution of the stack frame has failed.

- update ExecutionService to throw exceptions when execution goes wrong 
- update module names in the stack when the project name changes 

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/main/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/main/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guides.
- [x] All documentation and configuration conforms to the [markdown](https://github.com/enso-org/enso/blob/main/docs/style-guide/markdown.md) and [YAML](https://github.com/enso-org/enso/blob/main/docs/style-guide/yaml.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

File: engine/runtime/src/main/java/org/enso/interpreter/runtime/error/ConstructorDoesNotExistException.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package org.enso.interpreter.runtime.error;
```
Comment (kustosz) [inline engine/runtime/src/main/java/org/enso/interpreter/runtime/error/ConstructorDoesNotExistException.java]: This exception only makes sense in the context of the server, so please move it there.

File: engine/runtime/src/main/java/org/enso/interpreter/runtime/Context.java [hunk 2]
```diff
@@ -258,6 +258,18 @@ private void renamePackages(String oldName, String newName) {
     return getTopScope().getModule(moduleName);
   }
 
+  /**
+   * Fetches a module with a given name.
+   *
+   * @param moduleName the qualified name of the module to lookup.
+   * @return the relevant module.
+   */
+  public Module getModule(String moduleName) throws ModuleDoesNotExistException {
```
Comment (kustosz) [inline engine/runtime/src/main/java/org/enso/interpreter/runtime/Context.java]: Please move this to some support class that's associated with the execution server directly.
This API is already quite large, so one-user-module-specific helpers should be extracted.

**Review by kustosz — CHANGES_REQUESTED** [review]
Please move the execution-server-only-bits to execution-server-only packages. The types in `org.enso.interpreter.runtime` should only contain types directly relevant to the interpreter behavior.

**Comment by 4e6** [comment]
Created a separate `org.enso.interpreter.service.error` module for execution service errors.

**Review by kustosz — APPROVED** [review]

**Pull request merged**